language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/workspace/BallerinaWorkspaceManager.java

```diff
@@ -536,6 +554,109 @@ public String uriScheme() {
         return "file";
     }
 
+    @Override
+    public Optional<Process> run(Path filePath) throws IOException {
+        Optional<ProjectPair> projectPairOpt = projectPair(projectRoot(filePath));
+        if (projectPairOpt.isEmpty()) {
+            clientLogger.logWarning("Run command execution aborted because project is not loaded");
+            return Optional.empty();
+        }
+        ProjectPair projectPair = projectPairOpt.get();
+        if (!stopProject(projectPair)) {
+            clientLogger.logWarning("Run command execution aborted because couldn't stop the previous run");
+            return Optional.empty();
+        }
+
+        Project project = projectPair.project();
+        Package pkg = project.currentPackage();
+        Module executableModule = pkg.getDefaultModule();
+        PackageCompilation packageCompilation = pkg.getCompilation();
```

Comment:
Can be replaced by `waitAndGetPackageCompilation(filePath)`

Reply:
In the LS, compilation is only run  upto code analyzers by default(see https://discord.com/channels/957996897782616114/1106474013456347189). Therefore we have to try a project where other compiler phases are required and check the experience.